migrate/migrate.go

```diff
@@ -0,0 +1,251 @@
+package migrate
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/control-center/serviced/commons/docker"
+	"github.com/control-center/serviced/dao"
+	"github.com/control-center/serviced/datastore"
+	"github.com/control-center/serviced/domain/service"
+	"github.com/control-center/serviced/facade"
+	"github.com/control-center/serviced/utils"
+	"github.com/zenoss/glog"
+
+	dockerclient "github.com/zenoss/go-dockerclient"
+)
+
+const (
+	// The mount point in the service migration docker image
+	MIGRATION_MOUNT_POINT = "/migration"
+	// The well-known path within the service's docker image of the directory which contains the service's migration script
+	EMBEDDED_MIGRATION_DIRECTORY = "/opt/serviced/migration"
+)
+
+func RunMigrationScript(ctx datastore.Context, fcd *facade.Facade, request dao.RunMigrationScriptRequest) error {
```

Comment:
For ease of testability:

Instead of passing in the facade, create an interface that does only the requests you need.  For example, see https://github.com/control-center/serviced/blob/feature/hllb-release/scheduler/localsync.go